service/src/main/java/uk/gov/hmcts/reform/testingsupport/controllers/TestingSupportController.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.reform.testingsupport.controllers;
+
+import feign.FeignException;
+import io.swagger.annotations.Api;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.fpl.enums.State;
+import uk.gov.hmcts.reform.fpl.service.ccd.CoreCaseDataService;
+import uk.gov.hmcts.reform.testingsupport.service.TestingSupportService;
+
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.enums.State.PREPARE_FOR_HEARING;
+
+@Api
+@Slf4j
+@RestController
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@ConditionalOnProperty(prefix = "testing_support", name = "enabled", havingValue = "true")
+@SuppressWarnings("unchecked")
+public class TestingSupportController {
+    private static final String POPULATE_EVENT_ID_TEMPLATE = "populateCase-%s";
+    private final CoreCaseDataService coreCaseDataService;
+    private final TestingSupportService testingSupportService;
+
+    @PostMapping("/testingSupport/populateCase/{caseId}")
+    public void populateCase(@PathVariable("caseId") Long caseId, @RequestBody Map<String, Object> requestBody) {
```

Comment:
i would do this in e2e, This controller should only pass it to ccd. It will help with next step which is switching from e2e-fpl-ccd to e2e-ccd as data will be managed on e2e side and ready to reuse. Then TestSupportService is not needed.